torch_geometric/nn/conv/gprgnn.py

```diff
@@ -0,0 +1,167 @@
+from typing import Optional
+
+import numpy as np
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn import Parameter
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor, SparseTensor
+from torch_geometric.utils import spmm
+
+
+class GPRGNN(MessagePassing):
+    r"""The adaptive generalized PageRank layer from the
+    `"Adaptive Universal Generalized PageRank Graph Neural
+    Network" <https://arxiv.org/abs/2006.07988>`_ paper
+
+    .. math::
+        \mathbf{X}^{(0)} &= \mathbf{X}
+
+        \mathbf{X}^{(k)} &= \mathbf{\hat{D}}^{-1/2}
+        \mathbf{\hat{A}} \mathbf{\hat{D}}^{-1/2} \mathbf{X}^{(k-1)}
+
+        \mathbf{X}^{\prime} &= \sum_{k=0}^K \gamma_k \mathbf{X}^{(k)},
+
+    where :math:`\mathbf{\hat{A}} = \mathbf{A} + \mathbf{I}` denotes the
+    adjacency matrix with inserted self-loops and
+    :math:`\hat{D}_{ii} = \sum_{j=0} \hat{A}_{ij}` its diagonal degree matrix.
+    The adjacency matrix can include values other than :obj:`1`, representing
+    edge weights via the optional :obj:`edge_weight` tensor. Here
+    :math:`\gamma_k` denotes the learnable generalized PageRank (GPR)
+    weight for layer :math:`k`. There are four ways to initialize the
+    weights: :obj:`Delta`, Personalized PageRank (:obj:`PPR`), Negative
+    Personalized PageRank (:obj:`NPPR`), and :obj:`Random`.
+
+    Args:
+        K (int): Number of propagation layers :math:`K`.
+        alpha (float): Initialization parameter :math:`\alpha`.
+        init (str, optional): Initialization method for GPR weights.
```

Comment:
Can we list all the optional values of the `init` argument in the doc-string as well?

Reply:
Done.